cmd/earthly/root_cmds.go

```diff
@@ -488,33 +501,71 @@ func symlinkEarthlyToEarth() error {
 	return nil
 }
 
-func (app *earthlyApp) actionDocker(cliCtx *cli.Context) error {
-	app.commandName = "docker"
+func (app *earthlyApp) actionDockerBuild(cliCtx *cli.Context) error {
+	app.commandName = "docker-build"
 
-	dir := filepath.Dir(app.dockerfilePath)
-	earthfilePath := filepath.Join(dir, "Earthfile")
-	earthfilePathExists, err := fileutil.FileExists(earthfilePath)
+	flagArgs, nonFlagArgs, err := variables.ParseFlagArgsWithNonFlags(cliCtx.Args().Slice())
 	if err != nil {
-		return errors.Wrapf(err, "failed to check if %s exists", earthfilePath)
+		return errors.Wrapf(err, "parse args %s", strings.Join(cliCtx.Args().Slice(), " "))
 	}
-	if earthfilePathExists {
-		return errors.Errorf("earthfile already exists; please delete it if you wish to continue")
+	if len(nonFlagArgs) == 0 {
+		_ = cli.ShowAppHelp(cliCtx)
+		return errors.Errorf(
+			"no build context path provided. Try %s docker-build <path>", cliCtx.App.Name)
 	}
-	defer os.Remove(earthfilePath)
+	if len(nonFlagArgs) != 1 {
+		_ = cli.ShowAppHelp(cliCtx)
+		return errors.Errorf("invalid arguments %s", strings.Join(nonFlagArgs, " "))
+	}
+
+	buildContextPath := nonFlagArgs[0]
 
-	err = docker2earthly.Docker2Earthly(app.dockerfilePath, earthfilePath, app.earthfileFinalImage)
+	err = handleDockerFiles(buildContextPath)
 	if err != nil {
 		return err
 	}
 
-	fmt.Fprintf(os.Stderr, "Warning: earthly does not support all dockerfile commands and is highly experimental as a result, use with caution.\n")
+	argMap, err := godotenv.Read(app.argFile)
+	if err != nil && (cliCtx.IsSet(argFileFlag) || !errors.Is(err, os.ErrNotExist)) {
+		return errors.Wrapf(err, "read %q", app.argFile)
+	}
 
+	buildArgs, err := app.combineVariables(argMap, flagArgs)
+	if err != nil {
+		return errors.Wrapf(err, "combining build args")
+	}
+
+	content, err := docker2earthly.GenerateEarthfileContent(buildContextPath, app.dockerfilePath, app.earthfileFinalImage, buildArgs.Sorted(), app.platformsStr.Value(), app.dockerTarget)
+	if err != nil {
+		return errors.Wrap(err, "docker-build: failed to wrap Dockerfile with an Earthfile")
+	}
+
+	earthfilePath := filepath.Join(buildContextPath, "Earthfile")
+	defer func() {
+		os.Remove(earthfilePath)
+		os.Remove(filepath.Join(buildContextPath, ".earthlyignore"))
+	}()
+
+	out, err := os.Create(earthfilePath)
```

Comment:
this too feels odd -- the description of `docker-build` says `Build a Dockerfile without an Earthfile`.

If we must create any files, we should do so in a temporary directory and not in the user's CWD (unless the user has explicitly requested the temporary files to be kept).

Reply:
The files are removed in the end of execution, but I can switch it to use a temp dir instead

Reply:
I moved the temporary Earthfile to a temp dir (which is removed at the end of the execution).
If there is a .dockerignore file present in the build context,  a temporary .earthlyignore file is created in the build context dir - I was not able to get it to work if .earthlyignore is in the temp dir

Reply:
we not longer need to create .earthlyignore because FROM DOCKER would use existing .dockerignore files from the build context directory